spec/unit/backend/hocon_backend_spec.rb

```diff
@@ -0,0 +1,134 @@
+require 'spec_helper'
+require 'hiera/backend/hocon_backend'
+
+class Hiera
+  module Backend
+    class FakeCache
+      attr_accessor :value
+      def read(path, expected_type, default, &block)
+        read_file(path, expected_type, &block)
+      rescue => e
+        default
+      end
+
+      def read_file(path, expected_type, &block)
+        output = block.call(@value)
+        if !output.is_a? expected_type
+          raise TypeError
+        end
+        output
+      end
+    end
+
+    describe Hocon_backend do
+      before do
+        Config.load({})
+        Hiera.stubs(:debug)
+        Hiera.stubs(:warn)
+        @cache = FakeCache.new
+        @backend = Hocon_backend.new(@cache)
+      end
+
+      describe "#initialize" do
+        it "should announce its creation" do # because other specs checks this
+          Hiera.expects(:debug).with("Hiera HOCON backend starting")
+          Hocon_backend.new
+        end
+      end
+
+      describe "#lookup" do
+        it "should pick data earliest source that has it for priority searches" do
+          Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).yields(["one", "/nonexisting/one.hocon"])
+          @cache.value = "key: answer"
+
+          expect(@backend.lookup("key", {}, nil, :priority, nil)).to eq("answer")
+        end
+      end
+
+      describe "handling unexpected HOCON values" do
+          before do
+            Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).yields(["one", "/nonexisting/one.hocon"])
+          end
+
+          it "throws :no_such_key when key is missing in HOCON" do
+            @cache.value = "{}"
+            expect { @backend.lookup("key", {}, nil, :priority, nil) }.to throw_symbol(:no_such_key)
+          end
+
+          it "returns nil when the HOCON value is nil" do
+            @cache.value = "{key: null}"
+            expect(@backend.lookup("key", {}, nil, :priority, nil)).to be_nil
+          end
+
+          it "throws :no_such_key when the HOCON file is false" do
+            @cache.value = ""
+            expect { @backend.lookup("key", {}, nil, :priority, nil) }.to throw_symbol(:no_such_key)
+          end
+
+          it "raises a TypeError when the HOCON value is not a hash" do
+            @cache.value = "a: [one, two, three]"
+            expect { @backend.lookup("key", {}, nil, :priority, nil) }.to raise_error(TypeError)
+          end
+      end
+
+      it "should build an array of all data sources for array searches" do
+        Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).multiple_yields(["one", "/nonexisting/one.hocon"], ["two", "/nonexisting/two.hocon"])
+        @cache.expects(:read_file).with("/nonexisting/one.hocon", Hash).returns({"key"=>"answer"})
```

Comment:
It's possible (likely?) that we'll end up wanting to use the `.conf` file extension for these files, since that's the extension that we use everywhere else that we currently use HOCON.  I added a comment on the Jira ticket to solicit input on that, so let's not change it just yet, but if you wanted to poke around to figure out what it would take, that wouldn't hurt.